### PR TITLE
feat(thermocycler-gen2): update hardware controller

### PR DESCRIPTION
**Pull request by pmoegenburg**
Updates hardware controller to use AsyncResponseSerialConnection

### CODE REVIEW

File: api/tests/opentrons/hardware_control/modules/test_hc_thermocycler.py [hunk 1]
```diff
@@ -34,6 +39,31 @@ async def subject(usb_port: USBPort) -> AsyncGenerator[modules.Thermocycler, Non
     await therm.cleanup()
 
 
+@pytest.fixture
+async def subject_mocked_driver(
+    usb_port: USBPort,
+) -> AsyncGenerator[modules.Thermocycler, None]:
+    """Test subject with mocked driver"""
+    driver_mock = mock.AsyncMock()
```
Comment (mcous) [inline api/tests/opentrons/hardware_control/modules/test_hc_thermocycler.py]: `mock_driver` should be a separate fixture that's injected into this fixture

```python
@pytest.fixture
def mock_driver() -> mock.AsyncMock:
    return mock.AsyncMock(spec=DRIVER_CLASS_HERE)

@pytest.fixture
async def subject_mocked_driver(
    usb_port: USBPort,
    mock_driver: mock.AsyncMock, 
) -> AsyncGenerator[modules.Thermocycler, None]:
    ...
```

File: api/tests/opentrons/hardware_control/modules/test_hc_thermocycler.py [hunk 2]
```diff
@@ -234,3 +264,64 @@ async def test_cycle_temperature(
...
+async def fake_get_plate_temperature(*args, **kwargs):
+    return PlateTemperature(current=50, target=50)
+
+
+async def fake_get_lid_temperature(*args, **kwargs):
+    return Temperature(current=50, target=50)
+
+
+async def fake_get_lid_status(*args, **kwargs):
+    return ThermocyclerLidStatus.OPEN
```
Comment (mcous) [inline api/tests/opentrons/hardware_control/modules/test_hc_thermocycler.py]: These one-line functions to return one-line value objects make these tests harder for me to read. Could the values objects be used directly, instead?

File: api/tests/opentrons/hardware_control/modules/test_hc_thermocycler.py [hunk 3]
```diff
@@ -234,3 +264,64 @@ async def test_cycle_temperature(
...
+    ],
+)
+async def test_sync_error_response_to_poller(
+    subject_mocked_driver: modules.Thermocycler,
+    mock_get_plate_temp,
+    mock_get_lid_temp,
+    mock_get_lid_status,
+):
+    """Test that poll after synchronous temperature response with error updates module live data and status."""
+    subject_mocked_driver._driver.get_plate_temperature.return_value = (
```
Comment (mcous) [inline api/tests/opentrons/hardware_control/modules/test_hc_thermocycler.py]: One should never mock out implementation details of a test subject like this. This test should not know that `modules.Thermocycler` has a private `_driver` attribute.

Instead, set the mocks on the driver you're injecting (see comment above about creating a `mock_driver` fixture):

```python
async def test_sync_error_response_to_poller(
    mock_driver: mock.AsyncMock,
    subject_mocked_driver: modules.Thermocycler,
    ...,
):
    ...
    mock_driver.get_plate_temperature.return_value = ( ... )
```

**Review by mcous — CHANGES_REQUESTED** [review]

**Comment by sanni-t** [comment]
I think you want to target this to `edge`

**Comment by sfoster1** [comment]
> I think you want to target this to `edge`

No, unfortunately we're continuing to try and develop the TCGEN2 release independent of edge

**Comment by sfoster1** [comment]
Okay, we should target this on edge and once merged we'll cherry-pick to https://github.com/Opentrons/opentrons/pull/11607

**Comment by sfoster1** [comment]
Or we can target it on https://github.com/Opentrons/opentrons/pull/11607 and it'll come back in when we merge that back

**Comment by sfoster1** [comment]
Retargeting to `edge` since everything else for TCGEN2 is in there, and we're cherrypicking anyway.

**Review by mcous — APPROVED** [review]
Appreciate the test changes, LGTM

**Pull request merged**